projects/lib/src/board/westernboard.h

```diff
@@ -76,6 +76,33 @@ class LIB_EXPORT WesternBoard : public Board
 		/*! Movement mask for Rook moves. */
 		static const unsigned RookMovement = 8;
 
+		/*! Types of Pawn moves. */
+		enum StepType: const char
+		{
+			 NoStep = 0,      //!< Cannot move here
+			 FreeStep =  1,   //!< May move if target is empty
+			 CaptureStep = 2  //!< Capture opposing piece only
+			 /* FreeOrCaptureStep = FreeStep|CaptureStep //!< like King or Sergeant*/
+		};
+		/*! Stores a move \a type and a move direction \a file for a Pawn move */
+		struct PawnStep { StepType type; int file; };
+		/*!
+		 * Movement mask for Pawn moves.
+		 * Lists pawn move types and relative files
+		 * Forward left diagonal -1, straight 0, right diagonal 1.
+		 * \sa BerolinaBoard
+		 *
+		 * Default: A Pawn can step straight ahead onto a free square or
+		 * capture diagonally forward. So initialise this
+		 * as { {CaptureStep, -1}, {FreeStep, 0}, {CaptureStep, 1} }
+		 */
+		QList<PawnStep>  m_pawnSteps; 
```

Comment:
Yes, sorry. That hasn't been enough to get rid of the local copy. A drawback of using `QVarLengthArray` is that one cannot use an initializer list to fill in the values in one step. I tried it before launching the first pull request.  Elements must be added individually. I just made a version using `QVarLengthArray` with changes in constructor bodies and initialiser. 

I am testing now and will upload the `QVarLengthArray` version soon.

I also have a version using a `const QList<PawnStep> &` as target of the range-based loop.

1.) rename m_pawnSteps to m_pawnStepsList)
`QList<PawnStep> m_pawnStepsList;`

2.) use m_pawnSteps identifier, now for a const QList reference)
`const QList<PawnStep>& m_pawnSteps;`

3.) assignments are done in the `WesternBoard::WesternBoard` init list.
`: Board(zobrist),
m_pawnStepsList {{CaptureStep, -1}, {FreeStep, 0}, {CaptureStep, 1}},
m_pawnSteps(m_pawnStepsList),
...`

As I understand, this would also work without copying the QList in the range-based loops?